Pirat/Services/Resource/ResourceUpdateService.cs

```diff
@@ -132,24 +134,128 @@ public Task delete(string token)
             return Task.CompletedTask;
         }
 
-        public Task ChangeInformation(string token, Provider provider)
+        public Task<int> ChangeInformation(string token, Provider provider)
         {
-            return Task.CompletedTask;
+
+            AddressEntity location = new AddressEntity().build(provider.address);
+            _addressMaker.SetCoordinates(location);
+
+            var query = from o in _context.offer
+                join ap in _context.address on o.address_id equals ap.id
+                where o.token == token
+                select new {o, ap};
+
+            var l = query
+                .Select(x => x)
+                .ToList();
+            l.ForEach(collection =>
+            {
+                collection.o.name = provider.name;
+                //o.ispublic = provider.ispublic; //TODO everything non public so far
+                collection.o.organisation = provider.organisation;
+                collection.o.phone = provider.phone;
+                collection.ap.OverwriteWith(location);
+            });
+            
+            int changedRows = _context.SaveChanges();
+
+            if (2 < changedRows)
+            {
+                throw new InvalidDataStateException(Error.FatalCodes.UPDATES_MADE_IN_TOO_MANY_ROWS);
+            }
+
+            return Task.FromResult(changedRows);
         }
 
-        public Task ChangeInformation(string token, Consumable consumable)
+        public Task<int> ChangeInformation(string token, Consumable consumable)
         {
-            throw new NotImplementedException();
+
+            AddressEntity location = new AddressEntity().build(consumable.address);
+            _addressMaker.SetCoordinates(location);
+
+            var query = from o in _context.offer
+                join c in _context.consumable on o.id equals c.offer_id
+                join ac in _context.address on c.address_id equals ac.id
+                where o.token == token
+                select new {o, c, ac};
+
```

Comment:
If I am not mistaken, this function is going to change every consumable belonging to an offer? In the where-clause, a `c.id = consumable.id` is missing?